RobotURDF:  
  # Parameters
  parameters:
    rate: &rate 1000
    p_value: &k_p 100
    i_value: &k_i 0.1
    d_value: &k_d 10
    wheel_pid: &vel_pid {p: 5.0, i: 0.1, d: 0.0}
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: *rate
  # Effort Controllers -- Position and Velocity ----------------
  # Left Leg Controllers -----------------
  # Left Hip Controller
  LeftHip_controller:
    type: effort_controllers/JointEffortController
    joint: LeftHip_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Left Knee Controller
  LeftKnee_controller:
    type: effort_controllers/JointEffortController
    joint: LeftKnee_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Left wheel joint Controller
  LeftWheel_controller:
    type: effort_controllers/JointEffortController
    joint: LeftWheel_joint
    pid: *vel_pid
  # Front Leg Controllers -----------------
  # Front Hip Controller
  FrontHip_controller:
    type: effort_controllers/JointEffortController
    joint: FrontHip_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Front Knee Controller
  FrontKnee_controller:
    type: effort_controllers/JointEffortController
    joint: FrontKnee_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Front wheel joint Controller
  FrontWheel_controller:
    type: effort_controllers/JointEffortController
    joint: FrontWheel_joint
    pid: *vel_pid
  # Right Leg Controllers -----------------
  # Right Hip Controller
  RightHip_controller:
    type: effort_controllers/JointEffortController
    joint: RightHip_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Right Knee Controller
  RightKnee_controller:
    type: effort_controllers/JointEffortController
    joint: RightKnee_joint
    pid: {p: *k_p, i: *k_i, d: *k_d}
  # Right wheel joint Controller
  RightWheel_controller:
    type: effort_controllers/JointEffortController
    joint: RightWheel_joint
    pid: *vel_pid
